ultralytics/cfg/models/v10/yolov10x.yaml

```diff
@@ -0,0 +1,42 @@
+# Ultralytics YOLO 🚀, AGPL-3.0 license
+# YOLOv10 object detection model. For Usage examples see https://docs.ultralytics.com/tasks/detect
+
+# Parameters
+nc: 80 # number of classes
+scales: # model compound scaling constants, i.e. 'model=yolov8n.yaml' will call yolov8.yaml with scale 'n'
+  # [depth, width, max_channels]
+  x: [1.00, 1.25, 512]
```

Comment:
Can we combine sizes into a single YAML like YOLOv8?

Reply:
I'm going to prototype some ideas on this. What I have in mind might be best for a follow up PR.

Reply:
@Burhan-Q @glenn-jocher this might not be that easy as each model of yolov10 is slightly different i.e yolov10n vs yolov10s:
![pic-240526-2233-32](https://github.com/ultralytics/ultralytics/assets/61612323/f28c644a-2cdd-49e7-b130-4c19ce6dd49f)

Reply:
I was seeing this as well. I think that I can make a single config YAML still, but it will need to use a different structure to work. The biggest differences are in `n`, `s`, and `x`, scales; `m`, `b`, and `l` are essentially the same. 
- The `x` scale has one layer change relative to `m`, `b`, and `l`. 
- The `n` and `s` scales have 4 different layers and `n` has one layer different from `s`

Reply:
@Burhan-Q no need to work on a different structure for a third-party series of models for now I think.

Reply:
@Burhan-Q @Laughing-q ok got it guys, then yes leave all the YAMLs as different files.

Other ML developers don't really prioritize simplicity like we do unfortunately, so they'll take a well thought out implementation, like our single-architecture v8 YAML and produce 5x as many config files from it without thinking.